odoo/addons/base/ir/ir_qweb/qweb.py

```diff
@@ -308,8 +308,8 @@ def compile(self, template, options):
             raise e
         except Exception as e:
             path = _options['last_path_node']
-            node = element.getroottree().xpath(path)
-            raise QWebException("Error when compiling AST", e, path, etree.tostring(node[0], encoding='unicode'), name)
+            node = element.getroottree().xpath(path) if not ':' in path else None
```

Comment:
I think you have to re-read the template because the element changes when converting to a method. This may be the bug. The displayed nodes are therefore not correct even in 'normal' error cases

element, document = self.get_template(template, options)

Reply:
@Gorash 
Actually, that's not the bug.

When an xpath refers to a namespaced element, the method `element.xpath()` needs an additional parameter, which must be a dictionary mapping namespaces, in order to the method being able to translate them.

For instance:
```
>>> root = objectify.fromstring("""
...     <mynode:node xmlns:mynode="http://odoo.com/mynode"/>
... """)
>>> root.xpath("/mynode:node")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "src/lxml/etree.pyx", line 1577, in lxml.etree._Element.xpath
  File "src/lxml/xpath.pxi", line 307, in lxml.etree.XPathElementEvaluator.__call__
  File "src/lxml/xpath.pxi", line 227, in lxml.etree._XPathEvaluatorBase._handle_result
lxml.etree.XPathEvalError: Undefined namespace prefix
>>> root.xpath("/mynode:node", namespaces={'mynode': 'http://odoo.com/mynode'})
[<Element {http://odoo.com/mynode}node at 0x7f076ace0c88>]
>>> 
```

I'd like to add, the only possibility an xpath contains a ":" in that code, is when it's referencing to a namespaced element.